application/app/Http/Requests/UpdateInstitutionRequest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Rules\PhoneNumberRule;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+use Illuminate\Validation\Validator;
+
+class UpdateInstitutionRequest extends FormRequest
+{
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, ValidationRule|array|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'name' => [
+                'string',
+                'filled',
+            ],
+            'phone' => [
+                'nullable',
+                Rule::when(
+                    $this->input('phone') !== null,
+                    ['string', 'filled', new PhoneNumberRule]
+                ),
+            ],
+            'email' => [
+                'nullable',
+                Rule::when(
+                    $this->input('email') !== null,
+                    ['string', 'filled', 'email']
+                ),
+            ],
+            'short_name' => [
+                'nullable',
+                Rule::when(
+                    $this->input('short_name') !== null,
+                    ['string', 'filled', 'max:3']
+                ),
+            ],
```

Comment:
@DanBondarenko **[non-bloking]** looks like the code can be simplified a bit in a next way:

```php
    public function rules(): array
    {
        return [
            'name' => [
                'string',
                'filled',
            ],
            'phone' => [
                'nullable',
                'sometimes',
                'string',
                new PhoneNumberRule,
            ],
            'email' => [
                'nullable',
                'sometimes',
                'string', 'email'
            ],
            'short_name' => [
                'nullable',
                'sometimes',
                'string',
                'max:3',
            ],
        ];
    }
```

Reply:
Thank you for the suggestion. That indeed makes it simpler and I figured out the reason: the `ConvertEmptyStringsToNull` middleware means that any strings which are not `fillable` will be converted to `null`. Good to know in the future. I also found out that `sometimes` was actually redundant (at tests still passed) without it, so I skipped that one.